src/internal/observable/dom/AjaxObservable.ts

```diff
@@ -140,7 +161,11 @@ export class AjaxSubscriber<T> extends Subscriber<Event> {
       }
 
       // set headers
-      this.setHeaders(xhr, headers!);
+      for (const key in headers) {
+        if (headers.hasOwnProperty(key)) {
+          xhr.setRequestHeader(key, (headers as any)[key]);
```

Comment:
nitpick: with `headers` having a `Record` type, the `any` should not be necessary.